src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/BlobRouterMainTask.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.hmcts.reform.blobrouter.tasks;
+
+import com.azure.storage.blob.BlobServiceAsyncClient;
+import com.azure.storage.blob.models.BlobContainerItem;
+import org.slf4j.Logger;
+import uk.gov.hmcts.reform.blobrouter.config.ServiceConfiguration;
+import uk.gov.hmcts.reform.blobrouter.tasks.processors.ContainerProcessor;
+import uk.gov.hmcts.reform.blobrouter.util.StorageClientManager;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+public class BlobRouterMainTask {
+
+    static final String TASK_NAME = "blob-router";
+
+    private static final Logger LOGGER = getLogger(BlobRouterMainTask.class);
+
+    private final BlobServiceAsyncClient storageClient;
+    private final ContainerProcessor containerProcessor;
+    private final ServiceConfiguration serviceConfiguration;
+
+    public BlobRouterMainTask(
+        BlobServiceAsyncClient storageClient,
+        ContainerProcessor containerProcessor,
+        ServiceConfiguration serviceConfiguration
+    ) {
+        this.containerProcessor = containerProcessor;
+        this.serviceConfiguration = serviceConfiguration;
+        this.storageClient = storageClient;
+    }
+
+    public void run() {
+        LOGGER.info("Started {} job", TASK_NAME);
+
+        StorageClientManager
+            .getAvailableContainers(storageClient, serviceConfiguration)
+            .toStream()
+            .map(BlobContainerItem::getName)
+            .map(storageClient::getBlobContainerAsyncClient)
```

Comment:
Given we're just retrieving blob container client in order to immediately pass it to `containerProcessor`, I think it would be better to move `storageClient` to `ContainerProcessor` class from here, so that `containerProcessor` can get the right client itself.